src/auth/strategies/local.strategy.ts

```
@@ -12,4 +12,14 @@ export class LocalStrategy extends PassportStrategy(Strategy) {
       usernameField: "email",
     });
   }
+
+  async validate(username: string, password: string) {
+    const user = await this.authService.validateUser(username, password);
+
+    if (!user) {
+      throw new UnauthorizedException();
+    }
```

Don't think you need this. The validate method above should throw the appropriate error.